src/nexmark/queries/q19.rs

```diff
@@ -0,0 +1,289 @@
+use super::NexmarkStream;
+use crate::{
+    nexmark::model::{Bid, Event},
+    operator::{FilterMap, Fold},
+    Circuit, OrdZSet, Stream,
+};
+use std::collections::VecDeque;
+
+///
+/// Query 19: Auction TOP-10 Price (Not in original suite)
+///
+/// What's the top price 10 bids of an auction?
+/// Illustrates a TOP-N query.
+///
+/// ```sql
+/// CREATE TABLE discard_sink (
+///     auction  BIGINT,
+///     bidder  BIGINT,
+///     price  BIGINT,
+///     channel  VARCHAR,
+///     url  VARCHAR,
+///     dateTime  TIMESTAMP(3),
+///     extra  VARCHAR,
+///     rank_number  BIGINT
+/// ) WITH (
+///   'connector' = 'blackhole'
+/// );
+///
+/// INSERT INTO discard_sink
+/// SELECT * FROM
+/// (SELECT *, ROW_NUMBER() OVER (PARTITION BY auction ORDER BY price DESC) AS rank_number FROM bid)
+/// WHERE rank_number <= 10;
+/// ```
+
+type Q19Stream = Stream<Circuit<()>, OrdZSet<Bid, isize>>;
+
+const TOP_BIDS: usize = 10;
+
+pub fn q19(input: NexmarkStream) -> Q19Stream {
+    let bids_by_auction = input.flat_map_index(|event| match event {
+        Event::Bid(b) => Some((b.auction, b.clone())),
+        _ => None,
+    });
+
+    bids_by_auction
+        .aggregate::<(), _>(Fold::new(
+            VecDeque::with_capacity(TOP_BIDS),
+            |top: &mut VecDeque<Bid>, val: &Bid, _w| {
```

Comment:
Sorry, just noticed that the sorted behavior currently relies on the order of the `Bid` Model (it just so happened that my tests all used the same bidder, so it worked). Fixing...